dadi/lib/cache/index.js

```diff
@@ -95,8 +95,18 @@ Cache.prototype.getStream = function (key) {
 
   let encryptedKey = this.getNormalisedKey(key)
 
-  return cache.get(encryptedKey).catch(err => { // eslint-disable-line handle-callback-err
-    return null
+  return cache.get(encryptedKey).then(stream => {
```

Comment:
I'm not sure I'm a fan of changing the response format of this method to accommodate the need to return multiple values when requesting metadata. My preference would be to have a `getMetadata` method in the cache module, which would be called separately after an item has been successfully obtained from cache.

Instead of:

1. Call `cache.getStream(k)`, which returns an object `x`, and access the stream via `x.cachedStream` and the metadata via `x.metadata`

I would do:

1. Call `cache.getStream(k)` which returns a stream or `null`
1. If the above is truthy, I make a call for its metadata if I'm interested in it – `cache.getMetadata(k)`

This would mean no breaking changes to existing calls to `getStream`, and also you wouldn't need to drill down into the response object if you were only interested in the stream and not any metadata associated to it.